### PR TITLE
prov/efa: fix bugs in rnr support

### PR DESCRIPTION
**Pull request by wzamazon**
This PR fixes bugs in RNR support:

1. A handshake packet encounter RNR was dropped
2. Rx entry encountered RNR when sending RECEIPT packet was not queued thus RECEIPT packets was not resend
3. Tx entry encountered RNR when sending RTR packet was not queued thus RTR packets was not resend

### CODE REVIEW

File: prov/efa/src/efa.h [hunk 1]
```diff
@@ -620,11 +620,16 @@ void efa_rdm_peer_reset(struct rdm_peer *peer)
...
+	if (peer->flags & RXR_PEER_HANDSHAKE_QUEUED)
+		dlist_remove(&peer->handshake_queued_entry);
+
 	memset(peer, 0, sizeof(struct rdm_peer));
 #ifdef ENABLE_EFA_POISONING
 	rxr_poison_mem_region((uint32_t *)peer, sizeof(struct rdm_peer));
 #endif
 	dlist_init(&peer->rnr_entry);
+
+
```
Comment (bwbarrett) [inline prov/efa/src/efa.h:632]: why the blank lines on 631 and 632?

File: prov/efa/src/rxr/rxr_ep.c [hunk 2]
```diff
@@ -1673,17 +1673,23 @@ void rxr_ep_progress_internal(struct rxr_ep *ep)
 	 * Resend handshake packet for any peers where the first
 	 * handshake send failed.
 	 */
-	dlist_foreach_container_safe(&ep->peer_queued_list,
+	dlist_foreach_container_safe(&ep->handshake_queued_peer_list,
 				     struct rdm_peer, peer,
-				     queued_entry, tmp) {
+				     handshake_queued_entry, tmp) {
+
```
Comment (bwbarrett) [inline prov/efa/src/rxr/rxr_ep.c:1679]: it looks like this isn't new, but it doesn't seem like there should be a blank newline here?

File: prov/efa/src/rxr/rxr_pkt_type_misc.c [hunk 3]
```diff
@@ -98,41 +98,54 @@ ssize_t rxr_pkt_post_handshake(struct rxr_ep *ep, struct rdm_peer *peer)
...
-	 * avoid posting handshake packet multiple times to the
-	 * same peer.
-	 */
-	peer->flags |= RXR_PEER_HANDSHAKE_SENT_OR_QUEUED;
+
+	if (OFI_UNLIKELY(err)) {
+		FI_WARN(&rxr_prov, FI_LOG_EP_CTRL,
+			"Failed to post HANDSHAKE to peer %ld: %s\n",
+			peer->efa_fiaddr, fi_strerror(-err));
+		assert(0 && "Failed to post HANDSHAKE to peer");
```
Comment (bwbarrett) [inline prov/efa/src/rxr/rxr_pkt_type_misc.c:143]: Why is this assert here?  This does not seem to be a programming error, so either we should handle this case or we should abort always.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

File: prov/efa/src/efa.h (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/efa.h]: introduced by mistake. removed in new revision.

File: prov/efa/src/rxr/rxr_ep.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_ep.c]: removed the new line in new revision

File: prov/efa/src/rxr/rxr_pkt_type_misc.c (hunk 3 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_type_misc.c]: The assert was copied from previous implementation. I agree it is unnecessary. the next line `efa_eq_write_error()` handles the error properly. So I removed the assertion in new revision.

**Review by bwbarrett — APPROVED** [review]

**Pull request merged**